trie/database.go

```diff
@@ -99,6 +99,15 @@ type rawNode []byte
 func (n rawNode) cache() (hashNode, bool)   { panic("this should never end up in a live trie") }
 func (n rawNode) fstring(ind string) string { panic("this should never end up in a live trie") }
 
+func (n rawNode) EncodeRLP(w io.Writer) error {
+	count, err := w.Write([]byte(n))
+	if err != nil && count != len(n) {
+		err = fmt.Errorf("rawNode: could not write full length")
+	}
+
+	return err
+}
+
```

Comment:
Why is this needed?

Reply:
because otherwise it will add a `0x80+size` prefix when including it into the parent, which is redundant with the one already present in the `rawNode` at this stage. This in turn gives the wrong hash value for the parent. It only occurs when the RLP size is < 32

Reply:
But.. what makes the `w.Write` fail? And then you just wrap it in another error (using `Errorf` but no param to it)?